src/Residue.jl

```diff
@@ -16,9 +16,14 @@ function Residue(name::String)
     return Residue(handle)
 end
 
-function Residue(topology::Topology, index::Integer)
-    handle = lib.chfl_residue_from_topology(topology.handle, UInt64(index))
-    return Residue(handle)
+function Residue(topology::Topology, index::Integer; atom=false)
```

Comment:
The API seems a bit weird to me here:

```julia
topology = Topology()
# ...

residue = Residue(topology, 4)
residue = Residue(topology, 4; atom=true)
```

How about changing it to something like
```julia
# The constructor always get the nth residue in the topology
residue = Residue(topology, 4)
# Getting the optional residue for a given atom uses a separated function
residue = residue_for_atom(topology, 4)
```